File: rust/var-core/tests/hash_test.rs
//! Port of the FNV-1a vectors from `DriftTest.java` / `hash.test.ts`.

use var_core::hash::hash_source;

#[test]
fn hash_matches_the_typescript_vectors() {
    assert_eq!("fnv1a:4f9f2cab", hash_source("hello"));
    assert_eq!("fnv1a:1a47e90b", hash_source("abc"));
    assert_eq!("fnv1a:4eace75e", hash_source("# Title\n"));
}

File: go/core/hash_test.go
package varcore

import "testing"

// Port of the FNV-1a vectors from hash.test.ts / DriftTest.java.
func TestHashMatchesTypeScriptVectors(t *testing.T) {
	cases := []struct{ in, want string }{
		{"hello", "fnv1a:4f9f2cab"},
		{"abc", "fnv1a:1a47e90b"},
		{"# Title\n", "fnv1a:4eace75e"},
	}
	for _, c := range cases {
		if got := HashSource(c.in); got != c.want {
			t.Errorf("HashSource(%q) = %q, want %q", c.in, got, c.want)
		}
	}
}
